### PR TITLE
txn: Cover more corner cases for optimistic txn's tests

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

This PR does some minor changes for MvccTxn's tests.

* Cover the case that a delayed prewrite arrives later than committing.
* Cover the case that rollback is invoked when there is another transaction's lock.
* Merged `test_mvcc_txn_rollback` and `test_mvcc_txn_rollback_err` into one test.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

By CI

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No 

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -856,29 +844,44 @@ mod tests {
...
+        must_prewrite_lock(&engine, k, k, 10);
+        must_rollback(&engine, k, 10);
+        // data should be dropped after rollback
+        must_get_none(&engine, k, 20);
+
+        // Can't rollback committed transaction.
+        must_prewrite_put(&engine, k, v, k, 25);
+        must_commit(&engine, k, 25, 30);
+        must_rollback_err(&engine, k, 25);
+        must_rollback_err(&engine, k, 25);
```
Comment (sticnarf) [inline src/storage/mvcc/txn.rs]: What's for a repeated check?

Comment (MyonKeminta) [reply src/storage/mvcc/txn.rs]: Maybe trying to make sure the commit message is not removed, so the second rollback will still fail? I don't know. The old code did this too.
